### PR TITLE
feat(design-system): new icon set for status

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**

The new icon set is not yet in use.

**What is the chosen solution to this problem?**

Start with a simple component to prove the concept, and update `Status` with this new icon set

image

vs

image

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

Linked to https://jira.talendforge.org/browse/TUX-1173

### CODE REVIEW

File: packages/design-system/src/components/IconsProvider/IconsProvider.tsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
 import React, { ReactElement, RefObject, useState, useEffect, useRef } from 'react';
 import assetsAPI from '@talend/assets-api';
 
-const DEFAULT_BUNDLES = [assetsAPI.getURL('/dist/svg-bundle/all.svg', '@talend/icons')];
+const DEFAULT_BUNDLES = [
+	assetsAPI.getURL('/dist/svg-bundle/all.svg', '@talend/icons'),
+	assetsAPI.getURL('/dist/svg-bundle/XS.svg', '@talend/icons'),
+	assetsAPI.getURL('/dist/svg-bundle/S.svg', '@talend/icons'),
+	assetsAPI.getURL('/dist/svg-bundle/M.svg', '@talend/icons'),
+	assetsAPI.getURL('/dist/svg-bundle/L.svg', '@talend/icons'),
```
Comment (jmfrancois) [inline packages/design-system/src/components/IconsProvider/IconsProvider.tsx:9]: 😗

File: packages/design-system/src/components/Icon/SizedIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			<svg {...rest} style={{ width: numericSize, height: numericSize }} aria-hidden ref={ref}>
+				<use xlinkHref={`#${fullName}`} />
+			</svg>
+		);
+	},
+);
+
+SizedIcon.displayName = 'SizedIcon';
+
+export { SizedIcon };
```
Comment (Benrajalu) [inline packages/design-system/src/components/Icon/SizedIcon.tsx:30]: Why the object export? 

Also this doesn't seem exposed by the bundle.

File: packages/design-system/src/components/Icon/SizedIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React, { HTMLAttributes } from 'react';
+import { Icon, IconSize } from '@talend/icons';
+
+const getNumericSize = (size: IconSize) => {
+	return {
+		XS: 8,
+		S: 12,
+		M: 16,
+		L: 24,
```
[the remark this answers is not in the document]
Comment (frassinier) [reply packages/design-system/src/components/Icon/SizedIcon.tsx:9]: you're right!

File: packages/design-system/src/components/Icon/SizedIcon.tsx (hunk 2 above)
Comment (frassinier) [reply packages/design-system/src/components/Icon/SizedIcon.tsx:30]: Oh gosh! Well spotted!

**Review by Benrajalu — APPROVED** [review]

File: packages/design-system/src/components/Icon/SizedIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { Icon, IconSize } from '@talend/icons';
+
+const getNumericSize = (size: IconSize) => {
+	return {
+		XS: '0.8rem',
+		S: '1.2rem',
+		M: '1.6rem',
+		L: '2.4rem',
+	}[size];
+};
```
Comment (jsomsanith-tlnd) [inline packages/design-system/src/components/Icon/SizedIcon.tsx:11]: Instead of duplicating the values, should we use sass with token usage ?

**Review by jsomsanith-tlnd — APPROVED** [review]

File: packages/design-system/src/components/Icon/SizedIcon.tsx (hunk 4 above)
Comment (Benrajalu) [reply packages/design-system/src/components/Icon/SizedIcon.tsx:11]: We could

**Pull request merged**